File: Models/PlanEintrag.swift
import Foundation

struct PlanEintrag: Identifiable, Hashable {
    var id: String { key }
    let key: String
    let fach: String
    let klasse: String
    var thema: String
    var notiz: String
    var status: PlanStatus

    init(fach: String, klasse: String, thema: String = "", notiz: String = "", status: PlanStatus = .geplant) {
        self.key = "\(fach)_\(klasse)"
        self.fach = fach
        self.klasse = klasse
        self.thema = thema
        self.notiz = notiz
        self.status = status
    }
}

File: Windows-App/Unterrichtsplaner/Models/PlanEintrag.cs
using System.Text.Json.Serialization;

namespace Unterrichtsplaner.Models;

public class PlanEintrag
{
    public string Fach { get; set; } = "";
    public string Klasse { get; set; } = "";
    public string? Thema { get; set; }
    public string? Notiz { get; set; }
    public PlanStatus Status { get; set; } = PlanStatus.Geplant;

    [JsonIgnore]
    public string Key => $"{Fach}_{Klasse}";
}
